scripts/Switch.js

```diff
@@ -1,76 +1,27 @@
 export default function Switch() {
-  const buttonHome = getQuerySelector('home');
-  const buttonCreate = getQuerySelector('create');
-  const buttonBookmark = getQuerySelector('bookmark');
-  const buttonProfile = getQuerySelector('profile');
-
-  const pageHome = getQuerySelectorPage('home');
-  const pageBookmark = getQuerySelectorPage('bookmark');
-  const pageCreate = getQuerySelectorPage('create');
-  const pageProfile = getQuerySelectorPage('profile');
-
-  const headline = document.querySelector('[data-js="headline"]');
-
-  buttonHome.addEventListener('click', () => {
-    pageHome.classList.remove('hidden');
-    pageBookmark.classList.add('hidden');
-    pageCreate.classList.add('hidden');
-    pageProfile.classList.add('hidden');
-
-    buttonHome.classList.remove('footer--opacity');
-    buttonBookmark.classList.add('footer--opacity');
-    buttonCreate.classList.add('footer--opacity');
-    buttonProfile.classList.add('footer--opacity');
-
-    headline.innerText = 'QUIZ-APP';
-  });
-
-  buttonBookmark.addEventListener('click', () => {
-    pageHome.classList.add('hidden');
-    pageBookmark.classList.remove('hidden');
-    pageCreate.classList.add('hidden');
-    pageProfile.classList.add('hidden');
-
-    buttonHome.classList.add('footer--opacity');
-    buttonBookmark.classList.remove('footer--opacity');
-    buttonCreate.classList.add('footer--opacity');
-    buttonProfile.classList.add('footer--opacity');
-
-    headline.innerText = 'BOOKMARKS';
+  const allButtons = document.querySelectorAll('[data-nav]');
+  const allPages = document.querySelectorAll('[data-page]');
+
+  allButtons.forEach(button => {
+    button.addEventListener('click', event => {
+      allPages.forEach(page => {
+        page.classList.add('hidden'); //all pages hidden
+      });
+      allButtons.forEach(button => {
+        button.classList.add('footer--opacity'); //all buttons have opacity
+      });
+      const buttonAttribute = event.target.getAttribute('data-img');
+      const currentPage = document.querySelector(
+        `[data-page="${buttonAttribute}"]`
+      );
+      const currentButton = document.querySelector(
+        `[data-nav="${buttonAttribute}"]`
+      );
+      console.log(currentButton);
+      const heading = document.querySelector('h1');
```

Comment:
Your code looks very clear and fully functional, i think you dont need this console.log between any more